data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/ProcessorSwapPipelineIT.java

```diff
@@ -0,0 +1,141 @@
+    void tearDown() {
+        LOG.info("Test tear down. Stop the test runner.");
+        dataPrepperTestRunner.stop();
+    }
+
+    @Test
+    void run_with_single_record() {
+        final String messageValue = UUID.randomUUID().toString();
+        final Event event = JacksonEvent.fromMessage(messageValue);
+        final Record<Event> eventRecord = new Record<>(event);
+
+        LOG.info("Submitting a single record.");
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, Collections.singletonList(eventRecord));
+
+        await().atMost(400, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> assertThat(inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER), not(empty())));
+
+        final List<Record<Event>> records = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+
+        assertThat(records.size(), equalTo(1));
+
+        assertThat(records.get(0), notNullValue());
+        assertThat(records.get(0).getData(), notNullValue());
+        assertThat(records.get(0).getData().get("message", String.class), equalTo(messageValue));
+        assertThat(records.get(0).getData().get("test1", String.class), equalTo("knownPrefix10"));
+        assertThat(records.get(0).getData().get("test1_copy_original", String.class), equalTo("knownPrefix10"));
+
+        // Dynamically swap the pipeline processors
+        LOG.info("Swapping the pipeline processors");
+        List<Processor> targetPipelineProcessors = getTargetPipelineProcessors();
+        dataPrepperTestRunner.swapProcessors(PIPELINE_NAME_IN_YAML, targetPipelineProcessors);
+
+        // Send one more event and assert against the updated processor behavior
+        LOG.info("Submitting another single record.");
+        final String updatedMessageValue = UUID.randomUUID().toString();
+        final Event updatedEvent = JacksonEvent.fromMessage(updatedMessageValue);
+        final Record<Event> updatedEventRecord = new Record<>(updatedEvent);
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, Collections.singletonList(updatedEventRecord));
+
+        await().atMost(400, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> assertThat(inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER), not(empty())));
+
+        final List<Record<Event>> updatedRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+
+        LOG.info("Asserting the updated processor behavior");
+        assertThat(updatedRecords.size(), equalTo(2));
+        Event processedOutput = updatedRecords.get(1).getData();
+
+        assertThat(processedOutput, notNullValue());
+        assertThat(processedOutput, notNullValue());
+        assertThat(processedOutput.get("message", String.class), equalTo(updatedMessageValue));
+        assertThat(processedOutput.get("test1", String.class), equalTo("knownUpdatedPrefix10"));
+        assertThat(processedOutput.get("test1_copy_updated", String.class), equalTo("knownUpdatedPrefix10"));
+
+    }
+
+
+    private List<Processor> getTargetPipelineProcessors() {
```

Comment:
We shouldn't be loading the plugins here. At this point you are only testing a small sliver of the functionality. Again, if we use the HTTP server, this should be unnecessary.